selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/SeleniumWebDriverHelper.java

```diff
@@ -704,11 +704,19 @@ public void waitTextEqualsTo(WebElement webElement, String expectedText) {
    * @param timeout waiting time in seconds
    */
   public void waitTextContains(WebElement element, String expectedText, int timeout) {
+    String[] actual = new String[1];
     webDriverWaitFactory
         .get(timeout)
+        .withMessage(
+            () -> {
+              return "expected \n'" + expectedText + "'\nbut was \n'" + actual[0] + "'\n";
```

Comment:
Great idea!

Reply:
I introduced this in order to see _why_ the teste failed. It makes fixing much more convenient.